presto-main/src/main/java/io/prestosql/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -904,84 +692,86 @@ private Void visitScanFilterAndProjectInfo(
                 scanNode = Optional.empty();
             }
 
-            String format = "[";
-            String operatorName = "- ";
+            String formatString = "[";
+            String operatorName = "";
             List<Object> arguments = new LinkedList<>();
 
             if (scanNode.isPresent()) {
                 operatorName += "Scan";
-                format += "table = %s, ";
+                formatString += "table = %s, ";
                 TableHandle table = scanNode.get().getTable();
                 arguments.add(table);
                 if (stageExecutionStrategy.isPresent()) {
-                    format += "grouped = %s, ";
+                    formatString += "grouped = %s, ";
                     arguments.add(stageExecutionStrategy.get().isScanGroupedExecution(scanNode.get().getId()));
                 }
             }
 
             if (filterNode.isPresent()) {
                 operatorName += "Filter";
-                format += "filterPredicate = %s, ";
+                formatString += "filterPredicate = %s, ";
                 arguments.add(filterNode.get().getPredicate());
             }
 
-            if (format.length() > 1) {
-                format = format.substring(0, format.length() - 2);
+            if (formatString.length() > 1) {
+                formatString = formatString.substring(0, formatString.length() - 2);
             }
-            format += "] => [%s]";
+            formatString += "]";
 
             if (projectNode.isPresent()) {
                 operatorName += "Project";
-                arguments.add(formatOutputs(projectNode.get().getOutputSymbols()));
-            }
-            else {
-                arguments.add(formatOutputs(filterNode.get().getOutputSymbols()));
             }
 
-            format = operatorName + format;
-            print(indent, format, arguments);
-            printPlanNodesStatsAndCost(indent + 2,
-                    Stream.of(scanNode, filterNode, projectNode)
-                            .filter(Optional::isPresent)
-                            .map(Optional::get)
-                            .toArray(PlanNode[]::new));
-            printStats(indent + 2, planNodeId, true, true);
+            List<PlanNodeId> allNodes = Stream.of(scanNode, filterNode, projectNode)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .map(PlanNode::getId)
+                    .collect(toList());
+
+            NodeRepresentation nodeOutput = addNode(node, operatorName, format(formatString, arguments.toArray(new Object[0])), allNodes, ImmutableList.of(), ImmutableList.of());
```

Comment:
https://github.com/prestodb/presto/pull/12306

Reply:
Thanks, will merge this in.